staging/src/k8s.io/client-go/tools/events/eventseries_test.go

```diff
@@ -138,7 +137,11 @@ func TestEventSeriesf(t *testing.T) {
 
 	createEvent := make(chan *eventsv1.Event)
 	updateEvent := make(chan *eventsv1.Event)
-	patchEvent := make(chan *eventsv1.Event)
+	// An extra patch event might be created by `refreshExistingEventSeries` between the above 2 tests.
+	// Since `refreshExistingEventSeries` will lock the broadcaster and we don't consume the `patchEvent` channel in the second test,
+	// if channel `patchEvent` is not buffered, `refreshExistingEventSeries` would not release the lock that `recordToSink` try to
+	// acquire, so the create event would not be created during the second test and we got stuck.
+	patchEvent := make(chan *eventsv1.Event, 1)
```

Comment:
I agree with your analysis that you did to debug this. Based on stacktraces it seems to be exactly this.

Now I'm wondering if this is the best way to fix it.
Given that we don't really need `refreshExistingEventSeries`, maybe instead of this one we can actually disable this?

Maybe instead of doing `StartRecordingToSink` (we don't need neather refreshing nor finishing series in this test), we should simply start only appropriate parts of the broadcaster?

Reply:
SGTM. I have updated the test as you said.